controllers/sdkutil/clientwrapper/sagemaker_client.go

```diff
@@ -325,23 +347,65 @@ func (c *sageMakerClientWrapper) DeleteEndpointConfig(ctx context.Context, endpo
 
 	deleteResponse, deleteError := deleteRequest.Send(ctx)
 	if deleteError != nil {
-		if c.isDeleteEndpointConfig404Error(deleteError) {
-			//TODO: success and 404 both returns nil, nil
-			//      Add a mechanism to separate them.
-			return nil, nil
-		}
 		return nil, deleteError
 	}
 
 	return deleteResponse.DeleteEndpointConfigOutput, deleteError
 }
 
-// The SageMaker API does not conform to the HTTP standard. This detects if a SageMaker error response is equivalent
-// to an HTTP 404 not found.
-func (c *sageMakerClientWrapper) isDeleteEndpointConfig404Error(err error) bool {
+// The SageMaker API does not conform to the HTTP standard. The following methods detect
+// if a SageMaker error response is equivalent to an HTTP 404 not found.
+
```

Comment:
Remove newline, it breaks godocs. Can we copy the comment to the other functions?

Reply:
I moved this above all the methods to claim that the comment refers to all blocks of code below. As there eventually will be > 24 methods with the same header, I didn't want to copy/paste to each one every time the comment is changed.

Reply:
I wonder how other go packages do this?

Wow, generics would be so nice.